doc: add SDLC to design guide (#845)
rackerlabs/genestack: docs/sdlc.md

# Genestack SDLC

![SDLC](assets/images/sdlc.png){align=left : style="max-width:200px"}

> Software Development Life Cycle (SDLC) is the process used to ensure high quality solutions are delivered in a predictable repeatable method. The SDLC aims to produce outcomes that meets or exceeds customer expectations and reaches completion within time and cost estimates.

Reviewer: ```suggestion
Software Development Life Cycle (SDLC) is the process used to ensure high quality solutions are delivered in a predictable, repeatable fashion. The SDLC aims to produce outcomes that meet or exceed customer expectations and reaches completion within time and cost estimates.
```

> <br />The process is broken down into 6 distinct phases; Scope, Implement, Document, Test, Deployment, Maintain.  Each phase is outlined at a high level below.

Reviewer: ```suggestion
<br />The process is broken down into 6 distinct phases: Scope, Implement, Document, Test, Deployment, Maintain.  Each phase is outlined at a high level below.
```

### Scope

The scope phase is where new work is identified, based off the stated objective, and the level of effort determined. The plan portion of scope is where the work is assigned to various sprints to ensure timely completion. This step is vital to ensure we are meeting stated goals for the business and the community.

### Implement

In the implement phase, development teams use the requirements gathered in the scope phase to create code or process that meets the deliverable ask.

### Document

Documentation must reflect the current state of the codebase for any deployed application, service or process. If functionality has been added, removed, or changed the documentation is updated to reflect the change.

> Tl;dr changed something, added something, removed something document it.

Reviewer: ```suggestion
Tl;dr changed something, added something, removed something? Document it.
```

> Tl;dr changed something, added something, removed something document it.

Reviewer 2: ```suggestion
Tl;dr changed something, added something, removed something -- document it.
```

### Test

> The test phase is used to ensure that the deliverable is free from defects and meets the specified requirements. This is accomplished via a three-step or phased approach; github pre-commit checks, unit testing against development environment, and functional checks in a staging environment using rally against our Development and Staging environments.

Reviewer: ```suggestion
The test phase is used to ensure that the deliverable is free from defects
and meets the specified requirements. This is accomplished via a three-step or
phased approach:

1. Github pre-commit checks
2. Unit testing against development environment
3. Functional checks using [Rally](https://opendev.org/openstack/rally)
   against our Development and Staging environments
```
